### PR TITLE
Implemented "gradebook add" command

### PR DESCRIPTION
**Pull request by tristyxxnana**

### CODE REVIEW

File: docs/UserGuide.adoc [hunk 1]
```diff
@@ -465,18 +465,18 @@ Format: `gradebook add c/MODULE_CODE g/GRADE_ITEM [m/MARKS_MAX] [p/GRADE_PERCENT
 Examples:
 
 * `gradebook add c/CS2040C g/Assignment 1 m/10 p/10` +
-Adds grade component, Assignment 1, to module CS2040C with a maximum of 10 marks and constitutes 10% of overall marks.
+Adds grade gradebookComponent, Assignment 1, to module CS2040C with a maximum of 10 marks and constitutes 10% of overall marks.
 * `gradebook add c/CS2040C g/Final Examinations` +
-Adds grade component, Final Examinations, to module CS2040C.
+Adds grade gradebookComponent, Final Examinations, to module CS2040C.
```
Comment (harriuscai) [inline docs/UserGuide.adoc]: Every occurence of `component` here has been changed to `gradebookComponent

File: src/main/java/seedu/address/model/gradebook/GradebookComponent.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import javax.xml.bind.annotation.XmlElement;
+import javax.xml.bind.annotation.XmlRootElement;
+
+/**
+ * Represents a component in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+@XmlRootElement(name = "gradebook")
+
+public class GradebookComponent {
```
Comment (harriuscai) [inline src/main/java/seedu/address/model/gradebook/GradebookComponent.java]: Can you remove the newline between the annotation (`@XmlRootElement`) and the class declaration `public class GradebookComponent`?

File: src/main/java/seedu/address/model/gradebook/GradebookComponent.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    @XmlElement(name = "moduleCode")
+    public String getModuleCode() {
+        return moduleCode;
+    }
+
+    public void setModuleCode(String moduleCode) {
+        this.moduleCode = moduleCode;
+    }
+}
```
Comment (harriuscai) [inline src/main/java/seedu/address/model/gradebook/GradebookComponent.java]: Can you standardize the whitespacing between each method? Either no newline between methods, or one newline between methods :)

File: src/main/java/seedu/address/model/gradebook/GradebookComponent.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+    public GradebookComponent() {
+
+    }
+    public GradebookComponent(String gradeItemName, String moduleCode) {
+        this.gradeItemName = gradeItemName;
+        this.moduleCode = moduleCode;
+    }
+    @XmlElement(name = "gradeItemName")
+    public String getGradeItemName() {
```
Comment (harriuscai) [inline src/main/java/seedu/address/model/gradebook/GradebookComponent.java]: Move the 2 annotations `@XmlElement` from the getters to the variable declarations above

File: src/main/java/seedu/address/model/gradebook/GradebookComponent.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Represents a component in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+@XmlRootElement(name = "gradebook")
+
+public class GradebookComponent {
+
+    private String gradeItemName;
+    private String moduleCode;
```
Comment (harriuscai) [inline src/main/java/seedu/address/model/gradebook/GradebookComponent.java]: Annotate the 2 variables here like so: `@XmlElement(name = "<variableName>", required = true, nillable = true)`

File: src/main/java/seedu/address/model/gradebook/UniqueComponentList.java [hunk 6]
```diff
@@ -1,85 +1,37 @@
 package seedu.address.model.gradebook;
```
Comment (m-aslam-mj2) [inline src/main/java/seedu/address/model/gradebook/UniqueComponentList.java]: Is this file required?
